File: go/pgx/test/alternatives/manual_token/example_test.go
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

package manual_token_test

import (
	"os"
	"testing"

	manual_token "github.com/aws-samples/aurora-dsql-samples/go/pgx/src/alternatives/manual_token"
)

func TestManualTokenExample(t *testing.T) {
	if os.Getenv("CLUSTER_ENDPOINT") == "" {
		t.Skip("CLUSTER_ENDPOINT required for integration test")
	}

	err := manual_token.Example()
	if err != nil {
		t.Errorf("Example failed: %v", err)
	}
}

File: ruby/ruby-pg/test/alternatives/manual_token/example_test.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

require "rspec"
require_relative "../../../src/alternatives/manual_token/example"

RSpec.describe "manual_token example" do
  before(:all) do
    raise "CLUSTER_ENDPOINT is required" unless ENV["CLUSTER_ENDPOINT"]
    raise "CLUSTER_USER is required" unless ENV["CLUSTER_USER"]
    raise "REGION is required" unless ENV["REGION"]
  end

  it "runs the manual token example without error" do
    expect { main }.not_to raise_error
  end
end
